File: Leetcode/496-next-greater-element-i/next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();
        vector<int> ans;

        for(int i = 0;i<n;i++)
        {
            int target = nums1[i]; 
            auto it = find(nums2.begin(), nums2.end(), target);
            int index = it-nums2.begin();
            int a = -1;
            for(int j=index+1;j<m;j++)
            {
                if(target < nums2[j])
                {
                    a = nums2[j];
                    break;
                }
            }

            ans.push_back(a);
        }

        return ans;
    }
};

File: Leetcode/496-next-greater-element-i/next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        st = []
        nums2copy = nums2.copy()
        n1 = len(nums1)
        n2 = len(nums2)
        
        for i in range(n2-1,-1,-1):
            while st and st[-1] <= nums2[i]:
                st.pop()

            if not st:
                st.append(nums2[i])
                nums2copy[i] = -1
            else:
                nums2copy[i] = st[-1]
                st.append(nums2[i])
        # ans = []
        # for i in nums1:
        #     ans.append()
        ans = []
        for i in nums1:
            idx = nums2.index(i)
            ans.append(nums2copy[idx])

        return ans